src/tokens.md

```diff
@@ -328,6 +330,76 @@ b"\x52"; b"R"; br"R";                // R
 b"\\x52"; br"\x52";                  // \x52
 ```
 
+### C string and raw C string literals
+
+#### C string literals
+
+> **<sup>Lexer</sup>**\
+> C_STRING_LITERAL :\
+> &nbsp;&nbsp; `c"` ( ASCII_FOR_C_STRING | BYTE_ESCAPE | STRING_CONTINUE )<sup>\*</sup> `"` SUFFIX<sup>?</sup>
+>
+> ASCII_FOR_C_STRING :\
+> &nbsp;&nbsp; _any non-NUL ASCII (i.e 0x01 to 0x7F), except_ `"`, `\` _and IsolatedCR_
```

Comment:
This doesn't seem quite correct to me, since it looks like the parser allows a bare NUL in the string (there are later stages that perform the validation). The grammar specified in these sections needs to specify the format that the parser accepts. The prose below should explain any further restrictions placed on it.

Additionally, it looks like it does accept utf-8 input, so I'm not sure why it mentions the ascii restriction.

Reply:
Ah, interesting, I didn't realize the reference distinguished the grammar and permitted literal content. Fixed.

The ASCII restriction was just me misreading the RFC -- I didn't realize Unicode was permitted. I've fixed the notation so that it describes C string literals as accepting both byte escapes (like `b""`) and Unicode escapes (like `""`).